### DISCUSSION TITLE
Locating source of `observer` wrapped components [mobx-react]

### DISCUSSION THREAD

**Discussion by Silic0nS0ldier** [mobxjs/mobx#2939] [General]
I work with a large monorepo which uses Mobx extensively, with many (hundreds) of React components being wrapped with `observer`. This presents a challenge however as `observer` does a decent job of hiding the component implementation (going to source in React DevTools will just show the observer function source, particularly unhelpful when a function is anonymous).

If anyone has some tips on how to address this pain point, I'm all ears.

Further, if anyone has some ideas on how mobx might be refactored or DevTools enhanced to make the implementation more accessible (like how `React.memo` is treated) I'd be happy to discuss.

**Comment by iChenLei** [comment]
https://github.com/facebook/react/pull/21392
https://github.com/facebook/react/pull/20659
react memo's displayName issue in React Devtools is existing for a long time.
Hope the related pull request link is useful for you.

**Reply by BATCOH** [reply]
It seems that the question here is not about displayName, but about "View source of this element" devtools feature.
Even with correct displayName, it opens source of `observer` itself, not the source of wrapped component.

**Reply by iChenLei** [reply]
Thanks for your remind，it's really a annoyed issue. I will spend a little time to dig this question.
